### ISSUE TITLE
Sub-checker suppression performs insufficient traversal 

### ISSUE THREAD

**Issue by lazaroclapp** [uber/NullAway#391]
The method [symbolHasSuppressWarningsAnnotation](https://github.com/uber/NullAway/blob/e1d6960038806f8268bcb66ae655deb907998a51/nullaway/src/main/java/com/uber/nullaway/ErrorBuilder.java#L296) is used to check for fine grained suppressions of NullAway's sub-checkers, such as `@SuppressWarnings("NullAway.Init")` and `@SuppressWarnings("NullAway.Optional")`.

However, this method currently only checks for suppressions at the level of the specific `Tree` (technically: it's corresponding symbol) for which the error is emitted. This means it's not possible to suppress a sub-checker at the level of a whole class, a method containing multiple warnings, or a field or variable pointing to a lambda expression.

This is fine for the initialization sub-checker, since there is a very natural place where to suppress (the field itself), but an issue for optional nullness checking.

In particular, for code like:

```
class Foo {
    public static Function<Optional<T>,T> optionalGet = o -> o.get();
}
```

There is literally no way to suppress other than adding an `@SuppressWarnings(NullAway)`

If `T` is instead `Foo` with a method `@Nullable String foo()`, then there is no viable way of detecting the nullness dereference in the following code without globally disabling optional checking:

```
class Foo {
    public static Function<Optional<Foo>,String> bar = o -> o.get().foo().toString();
}
```

We need to make sub-checker suppressions traverse the. entire `TreePath`, at least checking: field, method, and class `Tree` nodes, before we can enable the optional checker by default.

**Comment by lazaroclapp** [comment]
@msridhar I wanted to turn on this checker internally and cut NullAway 0.8.0 with it enabled by default, but this seems like a blocker for that. Given that #389 is a crash, and we have a fix on master already, would you be fine with us testing internally without this option turned on and release as 0.7.10 still? 😦

**Comment by msridhar** [comment]
Yes, that is totally fine.  We can cut releases as needed while we fix the blockers around `Optional` checking.

**Comment by lazaroclapp** [comment]
This works at the method level, btw: https://github.com/uber/NullAway/blob/47f8d7b9eb0381c373839c0e8151a7f6b2142a14/nullaway/src/test/java/com/uber/nullaway/NullAwayTest.java#L1570

It's just no way to do this inside a lambda field. Haven't checked a lambda return value, etc.

**Comment by lazaroclapp** [comment]
I believe I fixed this in #392 some time ago. Closing.
